onlinejudge/_implementation/command/test.py

```diff
@@ -47,9 +48,18 @@ def f(x):
     return True
 
 
-def compare_and_report(proc: subprocess.Popen, answer: str, elapsed: float, memory: Optional[float], test_input_path: pathlib.Path, test_output_path: Optional[pathlib.Path], *, mle: Optional[float], mode: str, error: Optional[float], does_print_input: bool, silent: bool, rstrip: bool) -> str:
+def compare_and_report(proc: subprocess.Popen, answer: str, elapsed: float, memory: Optional[float], test_input_path: pathlib.Path, test_output_path: Optional[pathlib.Path], *, mle: Optional[float], mode: str, error: Optional[float], does_print_input: bool, silent: bool, rstrip: bool, judge: Optional[str]) -> str:
     # prepare the comparing function
-    if error:  # float mode
+    if judge is not None: # special judge mode
+        def match(command_output, _):
+            input = test_input_path.read_text()
+            stdin = tempfile.TemporaryFile()
+            stdin.write((input.rstrip('\n') + '\n' + command_output).encode('utf-8'))
+            stdin.seek(0)
+            info, proc = utils.exec_command(judge, stdin=stdin)
+            stdin.close()
+            return proc.returncode == 0
+    elif error:  # float mode
```

Comment:
```suggestion
    elif error is not None:  # float mode
```
これを試してみてください。
今更出てきた理由は不明ですが、、

Reply:
@fukatani 
`is not None`を試してみてもだめでした
試しにif文の条件分岐を変更して，`if error is not None`を先頭に持ってくると通ってしまったので，mypy の謎挙動なのではないかと推測しております．